sdk/include/wdf/kmdf/1.9/wdftypes.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * wdftypes.h
+ *
+ * Windows Driver Framework - C Driver Frameworks Basic Types
+ *
+ * This file is part of the ReactOS wdf package.
+ *
+ * Contributors:
+ *   Created by Benjamin Aerni <admin@bennottelling.com>
+ *
+ * Intended Usecase:
+ *   Kernel mode drivers
+ *
+ * THIS SOFTWARE IS NOT COPYRIGHTED
+ *
+ * This source code is offered for use in the public domain. You may
+ * use, modify or distribute it freely.
+ *
+ * This code is distributed in the hope that it will be useful but
+ * WITHOUT ANY WARRANTY. ALL WARRANTIES, EXPRESS OR IMPLIED ARE HEREBY
+ * DISCLAIMED. This includes but is not limited to warranties of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ */
+ #ifndef _WDFTYPES_H_
+ #define _WDFTYPES_H_
+
+
+
+#if (NTDDI_VERSION >= NTDDI_WIN2K)
+
+#define WDFAPI
+
+#ifndef __WDF_BYTE_DEFINED__
+#define __WDF_BYTE_DEFINED__
+typedef UCHAR BYTE;
+#endif
+
+#ifndef FORCEINLINE
+#define FORCEINLINE __forceinline
+#endif
+
+/* This definiton is used by WPP trace template file */
+#ifndef WDF_WPP_KMDF_DRIVER
+#define WDF_WPP_KMDF_DRIVER
+#endif
+
+typedef enum _WDF_TRI_STATE{
+    WdfFalse = FALSE,
+    WdfTrue = TRUE,
+    WdfUseDefault = 2,
+} WDF_TRI_STATE, *PWDF_TRI_STATE;
+
+typedef PVOID WDFCONTEXT;
+
+/* Declare structures needed by other headers */
+typedef struct WDFDEVICE_INIT *PWDFDEVICE_INIT;
+typedef struct _WDF_OBJECT_ATTRIBUTES PWDF_OBJECT_ATTRIBUTES;
+
+#define WDF_NO_OBJECT_ATTRIBUTES (NULL)
+#define WDF_NO_EVENT_CALLBACK (NULL)
+#define WDF_NO_HANDLE (NULL)
+#define WDF_NO_CONTEXT (NULL)
+#define WDF_NO_SEND_OPTIONS (NULL) 
+
+/* This is a generalized handle, should always be typeless */
+typedef HANDLE WDFOBJECT, *WDFOBJECT;
```

Comment:
Missing `P`.
This one's not a macro so the compiler should have caught it -- you can just include these headers from any kernel driver to make sure they compile.

Reply:
I actually haven't compiled anything with this *guilty look*